SampleBrowser/SampleBrowser/SceneDelegate.swift

```diff
@@ -13,15 +13,27 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
                willConnectTo session: UISceneSession,
                options connectionOptions: UIScene.ConnectionOptions) {
         guard let windowScene = (scene as? UIWindowScene) else { return }
-        let windowUUID = UUID()
-        let baseViewController = RootViewController(windowUUID: windowUUID)
-        window = UIWindow(windowScene: windowScene)
-        window?.rootViewController = baseViewController
-        window?.makeKeyAndVisible()
-
-        guard let window else { return }
-        let themeManager: ThemeManager = AppContainer.shared.resolve()
-        themeManager.setWindow(window, for: windowUUID)
-        themeManager.setSystemTheme(isOn: true)
+
+        let tempVC = UIViewController()
+        tempVC.view.backgroundColor = .systemBackground
+
+        let window = UIWindow(windowScene: windowScene)
+        self.window = window
+        window.rootViewController = tempVC
+        window.makeKeyAndVisible()
+
+        Task {
+            let engineProvider = await EngineProviderManager.shared.getProvider()
```

Comment:
This lines means the engine session and view gets created off the main thread, which is why this is breaking a lot of the assumptions in the `WebEngine`.